### PR TITLE
Detecting git url and using it as default

### PR DESCRIPTION
**Pull request by ganeshmannamal**

### CODE REVIEW

File: src/hubot-chatops.coffee [hunk 1]
```diff
@@ -21,7 +21,12 @@ module.exports = (robot) ->
   deploy = (repoName, branchName, res) ->
     room = if robot.adapterName == 'hipchat' then res.message.user.reply_to else res.message.room
 
-    repo = "https://github.com/#{repoName}.git"
+    if (repoName.match(/.*[:\/](.*)\/(.*).git/)){
+      repo = repoName
+    } else {
+      repo = "https://github.com/#{repoName}.git"
+    }
```
Comment (emilsoman) [inline src/hubot-chatops.coffee]: Will `repo` be available below, outside the if - else scope ?

Comment (ganeshmannamal) [inline src/hubot-chatops.coffee]: Now it defaults to gihub repo format

Comment (emilsoman) [inline src/hubot-chatops.coffee]: Will repo be available below, outside the if - else scope ?

Comment (ganeshmannamal) [inline src/hubot-chatops.coffee]: I am setting repo to gihub pattern first. Only if the regex matches it is changed. Am I missing something here?

Comment (emilsoman) [inline src/hubot-chatops.coffee]: I'm not able to see the code change which you are mentioning . Has the code been pushed ?

File: src/hubot-chatops.coffee [hunk 2]
```diff
@@ -20,8 +20,11 @@ module.exports = (robot) ->
   # a friendly message is posted to chat with this log URL.
   deploy = (repoName, branchName, res) ->
     room = if robot.adapterName == 'hipchat' then res.message.user.reply_to else res.message.room
-
     repo = "https://github.com/#{repoName}.git"
+
+    if repoName.match /.*[:\/](.*)\/(.*).git/
```
Comment (ganeshmannamal) [inline src/hubot-chatops.coffee]: @emilsoman this

**Comment by ganeshmannamal** [comment]
@emilsoman updated the pattern check

File: src/hubot-chatops.coffee [hunk 3]
```diff
@@ -42,7 +45,10 @@ module.exports = (robot) ->
   # the environment for a given github repo and branchname
   destroy = (repoName, branchName, res) ->
     room = if robot.adapterName == 'hipchat' then res.message.user.reply_to else res.message.room
-    repo = "https://github.com/#{repoName}.git"
+    repo = if (repoName.match(/.*[:\/](.*)\/(.*).git/))
+      repoName
+    else
+      "https://github.com/#{repoName}.git"
```
Comment (emilsoman) [inline src/hubot-chatops.coffee]: Can we DRY this up by extracting it into a function ?

Comment (ganeshmannamal) [inline src/hubot-chatops.coffee]: extracted to function

**Pull request merged**